bot/exts/help_channels/_cog.py

```diff
@@ -72,6 +73,9 @@ def __init__(self, bot: Bot):
 
         self.last_notification: t.Optional[datetime] = None
 
+        self.dynamic_message: t.Optional[int] = None
+        self.available_help_channels: t.Set[discord.TextChannel] = set()
```

Comment:
I think you don't need to keep the `dynamic_message` attribute on the class ~ you can just retrieve it from the cache every time you need it (inside the update function). Although this will have a little less overhead, so maybe it makes sense.

Regarding the `available_help_channels` attribute. In terms of code cleanliness I think it would be a lot nicer to always look at the `d.py` representation (as we do when the set isn't populated yet) rather than keeping our own state, but I understand that it may lag behind as `d.py` won't update the channels in the category when we send the request, but when some event comes back from Discord? So this is probably better, it just comes with the burden of having to make sure that we don't accidentally forget to update the set somewhere. Furthermore, it will desync if an admin moves a channel manually.